### PR TITLE
Adding randomized search.

### PR DESCRIPTION
**Pull request by jlevy44**
Going to test this code, tried to keep it as simple as possible. 

What are your thoughts?

### CODE REVIEW

**Comment by jlevy44** [comment]
Oh I may want to write some unit tests for these. Busy day at the lab for me.

File: hypopt/model_selection.py [hunk 1]
```diff
@@ -283,6 +289,8 @@ def fit(
             }
             params = list(ParameterGrid(self.param_grid))
             jobs = list(zip([job_params]*len(params), params))
+            if self.num_random_search != None:
+                jobs = random.sample(jobs,min(self.num_random_search,len(jobs)))
```
Comment (cgnorthcutt) [inline hypopt/model_selection.py:299]: Please put proper spacing/formatting. e.g. (a, b, c) not (a,b,c)

Comment (cgnorthcutt) [inline hypopt/model_selection.py:299]:
```suggestion
                jobs = random.sample(jobs, min(self.num_random_search, len(jobs)), )
```

File: hypopt/model_selection.py [hunk 2]
```diff
@@ -240,35 +246,35 @@ def fit(
 
         X_val : np.array of shape (n0, m)
             The validation data to optimize paramters with. If you do not provide this,
-            cross validation on the training set will be used. 
+            cross validation on the training set will be used.
```
Comment (cgnorthcutt) [inline hypopt/model_selection.py:255]: Most of your line changes aren't line changes?

**Review by cgnorthcutt — CHANGES_REQUESTED** [review]
Thanks for the PR. This is a great start. One could argue 'the right' way to do this is to create a new class called `RandomizedSearch` that has this functionality and matches the [sklearn interface](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html). This isn't something I intend to do, however it isn't much more beyond what you've already done and if you make a PR for it that includes corresponding test_randomizedsearch.py file in the [tests directory](https://github.com/cgnorthcutt/hypopt/tree/master/tests), I'll merge it.

File: hypopt/model_selection.py (hunk 2 above)
Comment (jlevy44) [reply hypopt/model_selection.py:255]: Yeah. I think my IDE reformatted some of these lines. I have no idea why this happened.

**Comment by jlevy44** [comment]
I completely agree that this should become a RandomizedSearch class. That's how I felt about this at the beginning, just have been pressed to complete research projects and wanted to flush this out quickly. I can go back and create a new inherited class.

**Comment by jlevy44** [comment]
I'll PR again with the proposed changes.

**Review by cgnorthcutt — COMMENTED** [review]
See: https://github.com/cgnorthcutt/hypopt/issues/10 - this is not what we want. The user should modify what data is shared, or protocol 4 should be used for pickling (works only in python 3.4+). Serial is NOT the way forward.

**Comment by cgnorthcutt** [comment]
> I'll PR again with the proposed changes.

Please do submit a pull request with randomized search as its own class. Happy to include that. For now closing this PR.

**Comment by jlevy44** [comment]
Sure apologies about this commit. Just did this to fix one of my experiments and forgot that it would update the PR. I’ll submit a RandomizedSearch as soon as work clears up, just been busy lately.

**Comment by prubbens** [comment]
Are there any updates on this?

**Pull request closed without merging**